project.go

```diff
@@ -281,19 +297,141 @@ func (c *ProjectRESTClient) DeleteUserMember(ctx context.Context, p *model.Proje
 	return handleSwaggerProjectErrors(err)
 }
 
+// AddMetadata adds metadata with a specific key and value to project p.
+// See this for more explanation of possible keys and values:
+// https://github.com/goharbor/harbor/blob/v1.10.2/api/harbor/swagger.yaml#L4894
+func (c *ProjectRESTClient) AddMetadata(ctx context.Context, p *model.Project, key ProjectMetadataKey, value string) error {
+	if p == nil {
+		return &ErrProjectNotProvided{}
+	}
+
+	m := &model.ProjectMetadata{}
+	switch key {
+	case EnableContentTrustProjectMetadataKey:
+		m.EnableContentTrust = value
+	case AutoScanProjectMetadataKey:
+		m.AutoScan = value
+	case SeverityProjectMetadataKey:
+		m.Severity = value
+	case ReuseSysCVEWhitelistProjectMetadataKey:
+		m.ReuseSysCveWhitelist = value
+	case PublicProjectMetadataKey:
+		m.Public = value
+	case PreventVulProjectMetadataKey:
+		m.PreventVul = value
+	}
+
+	_, err := c.parent.Client.Products.PostProjectsProjectIDMetadatas(&products.PostProjectsProjectIDMetadatasParams{
+		Metadata:  m,
+		ProjectID: int64(p.ProjectID),
+		Context:   ctx,
+	}, c.parent.AuthInfo)
+
+	err = handleSwaggerProjectErrors(err)
+	if err != nil {
+		t, ok := err.(*runtime.APIError)
+		if ok && t.Code == 409 {
+			// Unspecified error, which is returned when an existing metadata key
+			// is tried to create a second time.
+			return &ErrProjectMetadataAlreadyExists{}
+		}
+		return err
+	}
+
+	return handleSwaggerProjectErrors(err)
+}
+
+// GetMetadata retrieves metadata with key of project p.
+func (c *ProjectRESTClient) GetMetadata(ctx context.Context, p *model.Project, key ProjectMetadataKey) (string, error) {
```

Comment:
Here, the function name could be more descriptive as well, as it returns a specific metadata value:
```suggestion
// GetMetadataValue retrieves metadata with key of project p.
func (c *ProjectRESTClient) GetMetadataValue(ctx context.Context, p *model.Project, key ProjectMetadataKey) (string, error) {
```